core/services/workflows/store/models.go

```diff
@@ -0,0 +1,32 @@
+package store
+
+import "github.com/smartcontractkit/chainlink-common/pkg/values"
+
+const (
+	StatusStarted   = "started"
+	StatusErrored   = "errored"
+	StatusTimeout   = "timeout"
+	StatusCompleted = "completed"
+)
+
+type StepOutput struct {
+	Err   error
+	Value values.Value
+}
+
+type StepState struct {
+	ExecutionID string
+	Ref         string
+	Status      string
+
+	Inputs  *values.Map
+	Outputs *StepOutput
+}
+
+type ExecutionState struct {
```

Comment:
Sounds good -- I'm going to opt for WorkflowStep rather than ExecutionStep so that it maps with the table and makes it clear it's a subsidiary model of workflows.